lib/active_merchant/billing/gateways/payu_latam.rb

```diff
@@ -0,0 +1,393 @@
+      self.default_currency = "USD"
+      self.money_format = :dollars
+      self.supported_cardtypes = [:visa, :master, :american_express, :diners_club]
+
+      BRAND_MAP = {
+        "visa" => "VISA",
+        "master" => "MASTERCARD",
+        "american_express" => "AMEX",
+        "diners_club" => "DINERS"
+      }
+
+      def initialize(options={})
+        requires!(options, :merchant_id, :account_id, :api_login, :api_key)
+        super
+      end
+
+      def purchase(amount, payment_method, options={})
+        post = {}
+
+        add_credentials(post, 'SUBMIT_TRANSACTION')
+        add_transaction_type(post, 'AUTHORIZATION_AND_CAPTURE')
+        add_order(post, options)
+        add_buyer(post, options)
+        add_invoice(post, amount, options)
+        add_signature(post)
+        add_payment_method(post, payment_method)
+        add_payer(post, options)
+        add_browser_data(post, options)
+        add_extra_parameters(post, options)
+
+        commit('purchase', post)
+      end
+
+      def authorize(amount, payment_method, options={})
+        post = {}
+
+        add_credentials(post, 'SUBMIT_TRANSACTION')
+        add_transaction_type(post, 'AUTHORIZATION')
+        add_order(post, options)
+        add_buyer(post, options)
+        add_invoice(post, amount, options)
+        add_signature(post)
+        add_payment_method(post, payment_method)
+        add_payer(post, options)
+        add_browser_data(post, options)
+        add_extra_parameters(post, options)
+
+        commit('auth', post)
+      end
+
+      def capture(authorization, options={})
+        post = {}
+
+        add_credentials(post, 'SUBMIT_TRANSACTION')
+        add_transaction_type(post, 'CAPTURE')
+        add_reference(post, authorization)
+
+        commit('capture', post)
+      end
+
+      def void(authorization, options={})
+        post = {}
+
+        action, _, _ = split_authorization(authorization)
+
+        case action
+        when 'auth'
+          add_credentials(post, 'SUBMIT_TRANSACTION')
+          add_transaction_type(post, 'VOID')
+          add_reference(post, authorization)
+        when 'purchase', 'capture'
+          add_credentials(post, 'SUBMIT_TRANSACTION')
+          add_transaction_type(post, 'REFUND')
+          add_reference(post, authorization)
+        else
+          message = "Unsupported operation: successful Purchase, Authorize and unsettled Capture transactions can only be voided."
+          return Response.new(false, message)
+        end
+
+        commit('void', post)
+      end
+
+      # Gateway supports authorize for certain countries only,
+      # hence a combination of purchase and void for verify
+      #
+      # Each country has different minimum amounts,
+      # hence 100.00 instead of typical 1.00
+      def verify(credit_card, options={})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { purchase(10000, credit_card, options) }
```

Comment:
I'm thinking we should go with `authorize` here even if some countries don't support it.  We wouldn't want a `verify` to actually charge someone's card since  the `void` could fail.  If a country doesn't support `authorize`, then it doesn't support `verify`.